### PR TITLE
Ignore nested non-standard selectors in selector-max-specificity

### LINKED ISSUE #2684: Cannot parse selector error for at-each and nested variable interpolation in selector-max-specificity
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Bug where parser fails at interpolated SCSS as a class name.

> Which rule, if any, is this issue related to?

No specific rule, produces `Cannot parse selector` error.

> What CSS is needed to reproduce this issue?

```scss
@each $segment in $segments {
    .#{ map-get($segment, segmentPiece) } {
        .foo {
             border-color: map-get($segment, segmentPiece);
        }
    }
}
```

> What stylelint configuration is needed to reproduce this issue?

Seemingly any configuration, but using this:

```yaml
---
rules:
  at-rule-blacklist: ~
  at-rule-empty-line-before:
    - never
    -
      ignoreAtRules:
        - include
        - each
        - if
  at-rule-name-case: lower
  at-rule-name-newline-after: ~
  at-rule-name-space-after: always
  at-rule-no-vendor-prefix: true
  at-rule-semicolon-newline-after: always
  at-rule-whitelist: ~
  block-closing-brace-empty-line-before: never
  block-closing-brace-newline-after: always
  block-closing-brace-newline-before: always
  block-closing-brace-space-after: always-single-line
  block-closing-brace-space-before: always-single-line
  block-no-empty: true
  block-opening-brace-newline-after: always
  block-opening-brace-newline-before: ~
  block-opening-brace-space-after: always-single-line
  block-opening-brace-space-before: always
  color-hex-case: lower
  color-hex-length: short
  color-named: never
  color-no-hex: ~
  color-no-invalid-hex: true
  comment-empty-line-before: ~
  comment-no-empty: true
  comment-whitespace-inside: always
  comment-word-blacklist: ~
  custom-media-pattern: ~
  custom-property-empty-line-before: always
  custom-property-pattern: ~
  declaration-bang-space-after: never
  declaration-bang-space-before: always
  declaration-block-no-duplicate-properties: true
  declaration-block-no-shorthand-property-overrides: true
  declaration-block-properties-order: ~
  declaration-block-semicolon-newline-after: always
  declaration-block-semicolon-newline-before: ~
  declaration-block-semicolon-space-after: always-single-line
  declaration-block-semicolon-space-before: never
  declaration-block-single-line-max-declarations: 1
  declaration-block-trailing-semicolon: always
  declaration-colon-newline-after: ~
  declaration-colon-space-after: always-single-line
  declaration-colon-space-before: never
  declaration-empty-line-before: ~
  declaration-no-important: true
  declaration-property-unit-blacklist: ~
  declaration-property-unit-whitelist: ~
  declaration-property-value-blacklist: ~
  declaration-property-value-whitelist: ~
  font-family-name-quotes: always-where-recommended
  font-weight-notation: numeric
  function-blacklist: ~
  function-calc-no-unspaced-operator: true
  function-comma-newline-after: ~
  function-comma-newline-before: ~
  function-comma-space-after: always
  function-comma-space-before: never
  function-linear-gradient-no-nonstandard-direction: true
  function-max-empty-lines: 0
  function-name-case: lower
  function-parentheses-newline-inside: ~
  function-parentheses-space-inside: never
  function-url-data-uris: ~
  function-url-no-scheme-relative: ~
  function-url-quotes: never
  function-whitespace-after: always
  indentation: 2
  keyframe-declaration-no-important: true
  length-zero-no-unit: ~
  max-empty-lines: 1
  max-line-length: 90
  max-nesting-depth: 5
  media-feature-colon-space-after: always
  media-feature-colon-space-before: never
  media-feature-name-case: lower
  media-feature-name-no-vendor-prefix: true
  media-feature-parentheses-space-inside: never
  media-feature-range-operator-space-after: always
  media-feature-range-operator-space-before: always
  media-query-list-comma-newline-after: ~
  media-query-list-comma-newline-before: ~
  media-query-list-comma-space-after: always-single-line
  media-query-list-comma-space-before: never
  no-browser-hacks: ~
  no-descending-specificity: true
  no-duplicate-selectors: true
  no-empty-source: true
  no-eol-whitespace: ~
  no-extra-semicolons: true
  no-invalid-double-slash-comments: true
  no-missing-end-of-source-newline: ~
  no-unknown-animations: true
  no-unsupported-browser-features: ~
  number-leading-zero: always
  number-max-precision: 4
  number-no-trailing-zeros: true
  property-blacklist: ~
  property-case: lower
  property-no-unknown: true
  property-no-vendor-prefix: true
  property-whitelist: ~
  rule-empty-line-before: always-multi-line
  selector-attribute-brackets-space-inside: never
  selector-attribute-operator-blacklist: ~
  selector-attribute-operator-space-after: never
  selector-attribute-operator-space-before: never
  selector-attribute-operator-whitelist: ~
  selector-attribute-quotes: always
  selector-class-pattern: ~
  selector-combinator-space-after: always
  selector-combinator-space-before: always
  selector-id-pattern: ~
  selector-list-comma-newline-after: always-multi-line
  selector-list-comma-newline-before: ~
  selector-list-comma-space-after: always-single-line
  selector-list-comma-space-before: never
  selector-max-compound-selectors: 5
  selector-max-empty-lines: 0
  selector-max-specificity: "1,3,1"
  selector-nested-pattern: ~
  selector-no-attribute: ~
  selector-no-combinator: ~
  selector-no-id: ~
  selector-no-qualifying-type: true
  selector-no-type: ~
  selector-no-universal: ~
  selector-no-vendor-prefix: true
  selector-pseudo-class-blacklist: ~
  selector-pseudo-class-case: lower
  selector-pseudo-class-no-unknown: true
  selector-pseudo-class-parentheses-space-inside: never
  selector-pseudo-class-whitelist: ~
  selector-pseudo-element-case: lower
  selector-pseudo-element-colon-notation: ~
  selector-pseudo-element-no-unknown: true
  selector-type-case: lower
  selector-type-no-unknown: true
  shorthand-property-no-redundant-values: true
  string-no-newline: true
  string-quotes: single
  time-min-milliseconds: 100
  unit-blacklist:
    - pt
    - in
    - pc
  unit-case: lower
  unit-no-unknown: true
  unit-whitelist: ~
  value-keyword-case:
    - lower
    -
      ignoreKeywords:
        - segmentSlug
        - segmentAccent
        - segmentColor
        - segmentOver
        - segmentButtonHover
  value-list-comma-newline-after: never-multi-line
  value-list-comma-newline-before: never-multi-line
  value-list-comma-space-after: always
  value-list-comma-space-before: never
  value-no-vendor-prefix: true
```

> Which version of stylelint are you using?

`7.8.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Grunt with 
```js
stylelint: {
    options: {
         syntax: 'scss'
    },
    dist: {
        src: [
            '<%= pkg.paths.sass %>/**/*.scss',
            '<%= pkg.paths.patterns %>/**/*.scss'
        ]
    }
},
```

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SCSS maps.

> What did you expect to happen?

No errors to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

e.g. "The following warnings were flagged:"

```shell
Cannot parse selector
```

### PR DESCRIPTION
**Pull request by jeddy3**
Closes #2684

### CODE REVIEW

File: lib/rules/selector-max-specificity/README.md [hunk 1]
```diff
@@ -12,6 +12,8 @@ Visit the [Specificity Calculator](https://specificity.keegan.st) for visual rep
 
 This rule ignores selectors with variable interpolation (`#{$var}`, `@{var}`, `$(var)`).
 
+This rule ignores all selectors, standard syntax or otherwise, nested within non-standard rule constructs e.g. `@each`.
```
Comment (hudochenkov) [inline lib/rules/selector-max-specificity/README.md]: I can't understand this sentence :(

**Comment by jeddy3** [comment]
> Can we also close #2677 with this PR?

I'm afraid not as we don't have a heuristic in `isStandardSyntaxSelector` yet to determine that `.myMixin(@a, @b).bar` is non-standard.

File: lib/rules/selector-max-specificity/README.md (hunk 1 above)
Comment (jeddy3) [inline lib/rules/selector-max-specificity/README.md]: I pulled this out as I think the preceding statement is good enough:

> This rule ignores selectors with variable interpolation (`#{$var}`, `@{var}`, `$(var)`).

**Comment by jeddy3** [comment]
> -   Fixed: `selector-max-specificity` now ignores nested non-standard selectors ([#2685](https://github.com/stylelint/stylelint/pull/2685)).

**Pull request merged**